File: cpp/ISP/step2/Switches.cpp
#pragma once
#define interface struct

interface Switches {
	virtual void StartEngine() = 0;
	virtual void ShutDownEngine() = 0;
	virtual void TurnRadioOn() = 0;
	virtual void TurnRadioOff() = 0;
	virtual void TurnCameraOn() = 0;
	virtual void TurnCameraOff() = 0;
};

File: python/ISP/step2/switches.py
from abc import ABC, abstractmethod

class Switches(ABC):
    @abstractmethod
    def start_engine(self): pass

    @abstractmethod
    def shutdown_engine(self): pass

    @abstractmethod
    def turn_radio_on(self): pass

    @abstractmethod
    def turn_radio_off(self): pass

    @abstractmethod
    def turn_camera_on(self): pass

    @abstractmethod
    def turn_camera_off(self): pass
